lib/SGListView.js

```diff
@@ -61,10 +63,81 @@ var SGListView = React.createClass({
         ref='nativeListView'
         renderScrollComponent={this.renderScrollComponent}
         renderRow={this.renderRow}
-        onChangeVisibleRows={this.onChangeVisibleRows} />
+        onScroll={this.handleScroll}
+      />
     );
   },
 
+  /**
+   * Executed on every (throttled) scroll event. I don't like that, but
+   * some quick timing indicates it handles the interval query in <1ms,
+   * so it's definitely not a bottleneck and should perform pretty well
+   * even for a large number of elements.
+   */
+  handleScroll: function(evt) {
+    //  Expand the range that we check so we catch both cells
+    //  that become visible and cells that become invisible:
+    //
+    //  previous interval:  |------------|
+    //                     y1p          y2p
+    //
+    //  current interval:           |------------|
+    //                             y1           y2
+    //
+    //  test interval:      |--------------------|
+    //                     yt1                  yt2
+    //
+    var scrollProperties = this.refs.nativeListView.scrollProperties
+    var y1 = scrollProperties.offset
+    var y2 = y1 + scrollProperties.visibleLength
+
+    // Expand the interval so that things are rendered before they come on screen:
+    y1 -= this.props.preemptiveLoadDistance
+    y2 += this.props.preemptiveLoadDistance
+
+    // See above illustration:
+    var yt1 = Math.min(this.y1p,y1)
+    var yt2 = Math.max(this.y2p,y2)
+
+    // Perform an interval tree query to find all cells that may become
+    // either hidden or visible:
+    this.layoutTree.queryInterval( yt1, yt2, (interval) => {
+      var cell = interval.cell
+
+      if( interval[1] < y1 || interval[0] > y2 ) {
```

Comment:
Please reference the style guide. There's If/else spacing diversion as well as annotating logic tests.